### PR TITLE
Replace Travis CI with CircleCI

### PR DESCRIPTION
**Pull request by arlina-espinoza**
This PR replaces the Travis CI integration with CircleCI. Currently only runs the code sniffer - the same job that is running in travis (kernel, functional and JS tests are commented out for now).
To do after this PR: Enable kernel kernel, functional and JS tests - fix any tests that aren't passing.

### CODE REVIEW

File: phpcs.xml.dist [hunk 1]
```diff
@@ -18,5 +18,15 @@
...
     <severity>0</severity>
   </rule>
+  <!-- Core doesn't conform to this standard.
+         @see: https://www.drupal.org/project/coder/issues/2803251
+         @see: https://www.drupal.org/project/drupal/issues/1158720
+         -->
+    <rule ref="Drupal.Commenting.FunctionComment.TypeHintMissing">
+      <severity>0</severity>
+    </rule>
+    <rule ref="Drupal.Commenting.GenderNeutralComment.GenderNeutral">
```
Comment (cnovak) [inline phpcs.xml.dist]: I believe we fixed all comments to conform to this standard, I would rather fix comment than add this.

File: .circleci/code-sniffer.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+if [ ! -f dependencies_updated ]
+then
+  ./update-dependencies.sh $1
+fi
+
+# Install dependencies and configure phpcs
+vendor/bin/phpcs --config-set installed_paths vendor/drupal/coder/coder_sniffer
+
+# Check coding standards
+vendor/bin/phpcs -p -s -n --colors --standard=modules/apigee_edge/phpcs.xml.dist --report=junit --report-junit=artifacts/phpcs/phpcs.xml modules/$1
```
Comment (cnovak) [inline .circleci/code-sniffer.sh:17]: Do we want to [keep phpmd and phpmetrics](https://github.com/deviantintegral/drupal_tests/blob/master/hooks/code-sniffer.sh#L13)?

**Review by shadcn — APPROVED** [review]
Woah this is a major DX win @arlina-espinoza Thank you. LGTM 👍 

Do we have follow up tickets for "To do after this PR: Enable kernel kernel, functional and JS tests - fix any tests that aren't passing."?

**Review by cnovak — CHANGES_REQUESTED** [review]
Some small requests, and agree w/Arshad that we need issues for remaining tasks.

File: .circleci/code-sniffer.sh (hunk 2 above)
Comment (arlina-espinoza) [reply .circleci/code-sniffer.sh:17]: I've re-enabled phpmd and phpmetrics, which will not throw errors, but give interesting insights into the project. For example:
- https://76-174000098-gh.circle-artifacts.com/0/var/www/html/artifacts/phpmd/index.html
- https://76-174000098-gh.circle-artifacts.com/0/var/www/html/artifacts/phpmetrics/index.html

File: phpcs.xml.dist (hunk 1 above)
Comment (arlina-espinoza) [reply phpcs.xml.dist]: Re-enabled the `TypeHintMissing` and `GenderNeutralComment` sniffs, and yes, they were passing for the module. :)

**Comment by arlina-espinoza** [comment]
> Woah this is a major DX win @arlina-espinoza Thank you. LGTM +1
> 
> Do we have follow up tickets for "To do after this PR: Enable kernel kernel, functional and JS tests - fix any tests that aren't passing."?

Added follow up issue to enable other tests here: https://github.com/apigee/apigee-edge-drupal/issues/341

@cnovak I've addressed the requested changes, could you review again, please?

**Review by cnovak — APPROVED** [review]
LGTM

**Pull request merged**